include/deal.II/base/numbers.h

```diff
@@ -41,7 +41,7 @@
 // This is problematic when calling non-constexpr functions in constexpr
 // functions. Hence, we need a way to annotate functions explicitly as
 // host-only.
-#if defined(__clang__) && defined(__CUDA__)
+#if defined(__clang__) && defined(__CUDA__) || defined(KOKKOS_ENABLE_HIP)
```

Comment:
I always forget if `&&` has precedence over `||`.
```suggestion
#if (defined(__clang__) && defined(__CUDA__)) || defined(KOKKOS_ENABLE_HIP)
```
Also, the above command also doesn't match anymore. Maybe, say something like
```
// clang++ assumes that all constexpr functions are __host__ __device__ when `Kokkos` was configured with  `HIP` or `CUDA` support. [...]
```

Reply:
> I always forget if && has precedence over ||

It doesn't matter since all the HIP compiler are clang based